opencensus/trace/exporters/ocagent/trace_exporter.py

```diff
@@ -0,0 +1,189 @@
+from opencensus.trace.exporters import base
+from opencensus.trace.exporters.gen.opencensus.agent.common.v1 import (
+    common_pb2
+)
+from opencensus.trace.exporters.gen.opencensus.agent.trace.v1 import (
+    trace_service_pb2,
+    trace_service_pb2_grpc
+)
+from opencensus.trace.exporters.ocagent import utils
+from opencensus.trace.exporters.transports import sync
+
+# Default agent endpoint
+# TODO: subject to change: this is not a final default endpoint!
+DEFAULT_ENDPOINT = 'localhost:50051'
+
+# OpenCensus Version
+# TODO: https://github.com/census-instrumentation/opencensus-python/issues/296
+VERSION = '0.1.6'
+
+
+class TraceExporter(base.Exporter):
+    """Export the spans by sending them to opencensus agent.
+
+    :type service_name: str
+    :param service_name: name of the service
+
+    :type host_name: str
+    :param host_name: name of the host (machine or host name)
+
+    :type endpoint: str
+    :param endpoint: opencensus agent endpoint.
+
+    :type client: class:`~.trace_service_pb2_grpc.TraceServiceStub`
+    :param client: TraceService client stub.
+
+    :type transport: :class:`type`
+    :param transport: Class for creating new transport objects. It should
+                      extend from the base :class:`.Transport` type and
+                      implement :meth:`.Transport.export`. Defaults to
+                      :class:`.SyncTransport`. The other option is
+                      :class:`.BackgroundThreadTransport`.
+    """
+
+    def __init__(
+            self,
+            service_name,
+            host_name=None,
+            endpoint=None,
+            client=None,
+            transport=sync.SyncTransport):
+        self.transport = transport(self)
+
+        self.endpoint = DEFAULT_ENDPOINT if endpoint is None else endpoint
+
+        if client is None:
+            self.channel = grpc.insecure_channel(self.endpoint)
+            self.client = trace_service_pb2_grpc.TraceServiceStub(
+                channel=self.channel)
+        else:
+            self.client = client
+
+        self.node = common_pb2.Node(
+            identifier=common_pb2.ProcessIdentifier(
+                host_name=socket.gethostname() if host_name is None
+                else host_name,
+                pid=os.getpid(),
+                start_timestamp=utils.proto_ts_from_datetime(
+                    datetime.datetime.now())
+            ),
+            library_info=common_pb2.LibraryInfo(
+                language=common_pb2.LibraryInfo.Language.Value('PYTHON'),
+                version=VERSION
+            ),
+            service_info=common_pb2.ServiceInfo(name=service_name))
+
+    def emit(self, span_datas):
+        """
+        :type span_datas: list of :class:
+            `~opencensus.trace.span_data.SpanData`
+        :param list of opencensus.trace.span_data.SpanData span_datas:
+            SpanData tuples to emit
+        """
+
+        try:
+            responses = self.client.Export(
```

Comment:
Add a todo to add the optimization to keep the RPC alive. I got the answer from grpc team:
We need to use an iterator (an object that yields messages) similar to this example:
https://github.com/grpc/grpc/blob/master/examples/python/route_guide/route_guide_client.py#L65

No need to do it right now but good to have it documented.